provider/pkg/gen/schema.go

```diff
@@ -346,12 +346,32 @@ func (g *packageGenerator) genResource(typeName string, dd discoveryDocumentReso
 			if idPath == "" {
 				idPath = dd.getMethod.Path
 			}
-			v, err := g.buildIdParams(typeName, idPath, inputProperties, &response)
+			var queryParams strings.Builder
+			for param, details := range dd.getMethod.Parameters {
+				if details.Location != "query" || !details.Required {
+					continue
+				}
+				if queryParams.Len() == 0 {
+					_, err = queryParams.WriteString("?")
+					if err != nil {
+						return err
+					}
+				} else {
+					_, err = queryParams.WriteString("&")
+					if err != nil {
+						return err
+					}
+				}
+				queryParams.WriteString(fmt.Sprintf("%[1]s={%[1]s}", param))
+			}
+
+			params := queryParams.String()
+			v, err := g.buildIdParams(typeName, idPath, params, inputProperties, &response)
```

Comment:
It feels like it would be cleaner to append `params` to `idPath` and pass it as a single URL to `buildIdParams`. Would it be possible to amend `buildIdParams` to detect the name as the last parameter in the path, excluding query? Or change it to be the last unknown parameter (`projectId` would be known).

Just something to consider.

Reply:
See the latest update which should reflect the former suggestion you made. Let me know what you think.